examples-unsafe/src/curve25519/curve25519.rs

```diff
@@ -4,12 +4,14 @@ use hacspec_lib::*;
 public_nat_mod!(
     FieldElement,
     FieldCanvas,
+    FieldCanvasIdx,
```

Comment:
Hm, not sure if I like this. I also see it in `array`. What's the reason that we need it? A casual check shows that we always set it to `usize`.

Reply:
These are types which get transformed to refined types in F*, where we know that the `usize` is inside the bounds of the array. We need those to typecheck the specs out of the box in F*.

Reply:
Removed the indexes for `nat_mod` and `public_nat_mod`